addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1323,11 +1343,37 @@ MockServer.include({
      * @param {integer} message_id
      */
     _mockMailChannel_SetLastSeenMessage(ids, message_id) {
-        this._mockWrite('mail.channel', [ids, {
+        const cps = this._getRecords('mail.channel.partner', [['channel_id', 'in', ids], ['partner_id', '=', this.currentPartnerId]]);
+        this._mockWrite('mail.channel.partner', [cps.map(cp => cp.id), {
             fetched_message_id: message_id,
             seen_message_id: message_id,
         }]);
     },
+    /**
+     * Simulates `_inverse_channel_partner_ids` on `mail.channel`
+     *
+     * @private
+     */
+     _mockMailChannel_inverseChannelPartnerIds(channelIds) {
+        const channels = this._getRecords('mail.channel', [['id', 'in', channelIds]]);
+        for (const channel of channels) {
+            let outdatedCps = this._getRecords('mail.channel.partner', [['channel_id', '=', channel.id]]);
+            for (const memberId of channel.channel_partner_ids) {
+                const cp = this._getRecords('mail.channel.partner', [['partner_id', '=', memberId], ['channel_id', '=', channel.id]])[0];
+                if (!cp) {
+                    this._mockCreate('mail.channel.partner', {
+                        channel_id: channel.id,
+                        partner_id: memberId,
+                    });
+                } else {
+                    outdatedCps = outdatedCps.filter(cp => cp.partner_id !== memberId);
```

Comment:
could have domain for `outdatedCps` to be `partner_id not in channel.channel_partner_ids` instead?